Add unit tests for MapedSqlTokenParser and StringTokenParser

petersear opened the pull request:
Hi, I've analysed your codebase and seen some gaps in the coverage of:

org.yx.db.sql.token.MapedSqlTokenParser
org.yx.db.sql.token.StringTokenParser

I've written tests with the help of [Diffblue Cover](https://www.diffblue.com/products) and increased line coverage for these classes from 0% to approximately 50% (IntelliJ). Hopefully, they will help you detect regressions caused by future code changes.
I noticed that "skipTests" is set to "true" in pom.xml so this would need to change in order for these tests to run.

peterschrammel commented on src/test/java/org/yx/db/sql/token/MapedSqlTokenParserTest.java:
```diff
@@ -0,0 +1,40 @@
+  @Test
+  public void parseInputNotNullOutputNotNull() {
+    final MapedSql retval = new MapedSqlTokenParser("<", ">", null).parse("");
+    final ArrayList<Object> arrayList = new ArrayList<>();
```
move the arrayList to the assert that actually uses it. You could even inline the new.

petersear replied:
Thanks. Yes, I've inlined the new in each case.

peterschrammel reviewed (approved):
Diffblue Cover should be a link in the PR text

freddytuxworth commented on src/test/java/org/yx/db/sql/token/MapedSqlTokenParserTest.java:
```diff
@@ -0,0 +1,40 @@
+    Assert.assertNotNull(retval);
+    Assert.assertEquals("", retval.getSql());
+    Assert.assertNull(retval.getEvent());
+    Assert.assertEquals(arrayList, retval.getParamters());
```
I think in these tests instead of constructing a new empty `ArrayList` and comparing it with `retval.getParameters()`, you could just verify that `retval.getParameters().length() == 0`. That way you don't have to setup `arrayList`

petersear replied:
I've gone with keeping the arrayList to match what the code does and inlining the new as @peterschrammel suggested.

petersear commented:
Thanks, PR text updated to include link to Diffblue website, new ArrayList<>() inlined.

sudo-jaa commented on src/test/java/org/yx/db/sql/token/MapedSqlTokenParserTest.java:
```diff
@@ -0,0 +1,37 @@
+package org.yx.db.sql.token;
+
+import org.junit.Assert;
```
Provided this repo doesn't have any style requirement that prohibit this, I think it's preferably to import `org.junit.Assert.*;` and collapse your assertions down to, for example: `assertEquals`

sudo-jaa commented on src/test/java/org/yx/db/sql/token/MapedSqlTokenParserTest.java:
```diff
@@ -0,0 +1,37 @@
+
+public class MapedSqlTokenParserTest {
+
+  @Test
```
we may be able to collapse this down to a single method: 

```
@Test
  public void testParse() {
    final MapedSql parsed = new MapedSqlTokenParser("<", ">", null).parse("");
    assertNotNull(parsed);
    assertEquals("", parsed.getSql());
    assertNull(parsed.getEvent());
    assertEquals(new ArrayList<>(), parsed.getParamters());

    final MapedSql parsed2 = new MapedSqlTokenParser("<", ">", null).parse(null);
    assertNotNull(parsed2);
    assertEquals("", parsed2.getSql());
    assertNull(parsed2.getEvent());
    assertEquals(new ArrayList<>(), parsed2.getParamters());

    final MapedSql parsed3 = new MapedSqlTokenParser("<", ">", null).parse("text");
    assertNotNull(parsed3);
    assertEquals("text", parsed3.getSql());
    assertNull(parsed3.getEvent());
    assertEquals(new ArrayList<>(), parsed3.getParamters());
  }
```

petersear replied:
I've changed this to save a few lines. I was in two minds because that's 12 asserts in one test. If one assert fails the whole test fails which is maybe harder to track down than in more granular tests. I've gone with less is more though.
In the other class, StringTokenParser, I'd already combined the asserts into one test but I've now changed the method name to testParse() to match this one. Thanks.

Commit:
Add unit tests for MapedSqlTokenParser and StringTokenParser

Unit tests added for:
org.yx.db.sql.token.MapedSqlTokenParser
org.yx.db.sql.token.StringTokenParser

These tests were written using Diffblue Cover.
